bson/src/test/unit/org/bson/internal/ProvidersCodecRegistrySpecification.groovy

```diff
@@ -115,22 +115,32 @@ class ProvidersCodecRegistrySpecification extends Specification {
 
     def 'get should use the codecCache'() {
         given:
-        def provider = Mock(CodecProvider)
+        def codec = Mock(Codec)
+        def provider = new CodecProvider() {
+            private int counter = 0
+
+            @Override
+            Codec get(final Class clazz, final CodecRegistry registry) {
+                if (counter == 0) {
+                    counter++
+                    return codec
+                }
+                throw new AssertionError((Object)'Must not be called more than once.')
+            }
+        }
```

Comment:
For some reason, I failed to use [Spock stubbing](https://spockframework.org/spock/docs/2.0/all_in_one.html#_returning_fixed_values) here (the stabbed `provider` works as expected if used directly from the test, but does not work when used via `registry`. So I had to stub the object manually.